### PR TITLE
Ports cogbars from TG

### PR DESCRIPTION
**Pull request by Tsar-Salat**
## About The Pull Request

ezgif-5-8216ea8712

"Cog Progressbars" are basically a notification that someone is doing a noticeable do_after, that other players are able to see.

Examples being(nonexhaustive):
- Rolling dough with rollingpin
- Climbing a table
- Crafting
- Constructing
- CPR
- Creating a slimelink with someone as a xenobio slime race

Examples that, explicitly, others CANNOT see:
- Loading the nuke core into the core briefcase
- sucking blood progressbar as vampire race
- struggling out of cryo chamber
- Draining heretic influences
- Heretic Manse Link
- Chameleon gear
- Space Ninja draining actions

Ports:
- https://github.com/tgstation/tgstation/pull/52521
- https://github.com/tgstation/tgstation/pull/55172
- https://github.com/tgstation/tgstation/pull/77940
- https://github.com/tgstation/tgstation/pull/82413
- https://github.com/tgstation/tgstation/pull/82416

## Why It's Good For The Game

There is currently ZERO actual visual feedback that someone is preforming a do_after progressbar, even when it would make sense that they are. No, a `to_chat` does not count.

This should provide a simple way to display to others that someone is interacting with something, with an easy way to blacklist stuff that shouldn't be as obvious

## Testing Photographs and Procedure

<details>
<summary>Screenshots&Videos</summary>

Terribly edited, but it shows a few uses

https://github.com/BeeStation/BeeStation-Hornet/assets/62388554/e9387dc0-03fb-46a7-9d16-e0d05574cf41

</details>

## Changelog
:cl: rkz, jlsnow301, Jacquerel, Mothblocks
add: adds cogbars, a way for players to see others preforming actions
refactor: refactored CPR. You no longer have to click the patient for every pump. The action will repeat, slowly at first until gaining enough speed when your character starts getting in-tune with the Beegee's song.
refactor: refactored the way progressbars display on very large mobs, now they will be centre'd like expected
refactor: do_afters and numerous timed_action_flags
fix: npcs do not make progressbars, saves processing
/:cl:

### CODE REVIEW

**Comment by Fronsis** [comment]
coggers!!!

File: code/__DEFINES/layers.dm [hunk 1]
```diff
@@ -142,6 +142,8 @@
 #define BALLOON_CHAT_PLANE 651
 
 ///--------------- FULLSCREEN IMAGES ------------
+///Anything that wants to be part of the game plane, but also wants to draw above literally everything else
+#define HIGH_GAME_PLANE 499
```
Comment (Rukofamicom) [inline code/__DEFINES/layers.dm]: Why is this defined under the fullscreen images section?

Why is it drawn over the white from being flashed?

Comment (Tsar-Salat) [reply code/__DEFINES/layers.dm]: > Why is this defined under the fullscreen images section?

fixed in fc5f3de, good catch


> Why is it drawn over the white from being flashed?

That define is part of the fullscreen layer. 1, 2, 3, 4 are their own thing on fullscreen

**Comment by Rukofamicom** [comment]
Planning to TM this later today.

**Comment by Tsar-Salat** [comment]
Bugs found during TM:
- Cogbars appearing when they shouldnt (namely handcuff resisting)
- TGUI strip menu only allowing one strip at a time
- Runtimes in progressbar (though it was a clock round, and it occurred on clock gear, so unsure)

**Review by Rukofamicom — APPROVED** [review]

**Comment by Rukofamicom** [comment]
I don't see any particular further issues in testing.

**Pull request merged**